Database/TestDatabaseCache.php

```diff
@@ -0,0 +1,152 @@
+     *
+     * @return \DateTime|null
+     */
+    protected function getClassLastModified($class)
+    {
+        $lastModifiedDateTime = null;
+
+        $reflClass = new \ReflectionClass($class);
+        $classFileName = $reflClass->getFileName();
+
+        if (file_exists($classFileName)) {
+            $lastModifiedDateTime = new \DateTime();
+            $lastModifiedDateTime->setTimestamp(filemtime($classFileName));
+        }
+
+        return $lastModifiedDateTime;
+    }
+
+    /**
+     * Determine if the Fixtures that define a database backup have been
+     * modified since the backup was made.
+     *
+     * @param array $classNames
+     *            The fixture classnames to check
+     * @param string $backup
+     *            The fixture backup SQLite database file path
+     *
+     * @return bool TRUE if the backup was made since the modifications to the
+     *         fixtures; FALSE otherwise
+     */
+    private function isBackupUpToDate(array $classNames, $backup)
+    {
+        if(!file_exists($backup) || !file_exists($backup.'.ser')) {
+            return false;
+        }
+
+        $backupLastModifiedDateTime = new \DateTime();
+        $backupLastModifiedDateTime->setTimestamp(filemtime($backup));
+
+        foreach ($classNames as &$className) {
+            $fixtureLastModifiedDateTime = $this->getClassLastModified($className);
+            if ($backupLastModifiedDateTime < $fixtureLastModifiedDateTime) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     *
+     * @param array $metadatas
+     * @param array $classNames
+     * @return string full path to the database cache file
+     */
+    private function buildCacheFilePath(array $metadatas, array $classNames)
+    {
+        return $this->container->getParameter('kernel.cache_dir') . '/test_' . md5(serialize($metadatas) . serialize($classNames)) . '.db';
+    }
+
+    /**
+     * Get Filename of file base SQLite database
+     *
+     * @param array $params see Doctrine\DBAL\Connection->getParams()
+     * @throws \InvalidArgumentException
+     * @return string
+     */
+    public function getSQLiteName(array $params)
+    {
+        if (isset($params['master'])) {
+            $params = $params['master'];
+        }
+
+        $name = isset($params['path']) ? $params['path'] : (isset($params['dbname']) ? $params['dbname'] : false);
+        if (!$name) {
+            throw new \InvalidArgumentException("Connection does not contain a 'path' or 'dbname' parameter and cannot be dropped.");
+        }
+
+        return $name;
+    }
+
+    /**
+     * Is caching of SQLite Database instances enabled?
+     *
+     * Caching can be enabled in app/config/config_test.yml:
+     * ``` yaml
+     * liip_functional_test:
+     *     cache_sqlite_db: true
+     * ```
+     */
+    public function isCacheEnabled()
+    {
+        return $this->container->getParameter('liip_functional_test.cache_sqlite_db');
```

Comment:
instead of injecting the whole container to be able to retrieve 2 parameters from it, you should inject only these 2 scalars